src/solvers/imex_ssprk.jl

```diff
@@ -144,6 +148,9 @@ function step_u!(integrator, cache::IMEXSSPRKCache)
                 T_exp!(T_exp, U, p, t_exp)
             end
         end
+        apply_filter!(U, p, t_imp)
+        post_stage_callback!(U, p, t_imp)
+
```

Comment:
Do we need this additional call to the filter, and the corresponding one in `imex_ssprk.jl`? Based on the algorithm formulation, we should only need two calls to the filter for both ARK and SSPRK methods. Since DSS is fairly expensive, we should probably avoid it where it is not necessary.

Reply:
I suppose not?

Reply:
This is just a performance optimization, though, correct?

Reply:
Yes, adding an extra filter operation here probably would not change our results beyond roundoff error. This relies on the assumption that the implicit tendency preserves whatever constraint is enforced by the filter, which should be true for our combination of `T_imp!` and `dss!` (up to roundoff error).